pkg/volume/emptydir/empty_dir.go

```diff
@@ -331,11 +332,7 @@ func (ed *emptyDir) setupTmpfs(dir string) error {
 		return nil
 	}
 
-	var options []string
-	// Linux system default is 50% of capacity.
-	if ed.sizeLimit != nil && ed.sizeLimit.Value() > 0 {
-		options = []string{fmt.Sprintf("size=%d", ed.sizeLimit.Value())}
-	}
+	options := ed.generateTmpfsMountOptions(util.IsTmpfsNoswapOptionSupported(ed.mounter))
```

Comment:
should we only use `noswap` if swap is enabled in the OS?

Reply:
Definitely. Already working on that.

Reply:
@liggitt After finishing the implementation - I'm second guessing this statement :)

One corner case is if a user would start kubelet and only then create a swap file. kubelet currently does not detect if that's happening. The admin can restart kubelet, but then I guess the already mounted volumes would lack the `noswap` option.

I'm not sure if there's any reason not to use the `noswap` option if it's supported. I even think that in the future, when it is widely supported enough, we can drop this logic and use the `tmpfs` option indefinitely.

WDYT?

Reply:
> @liggitt After finishing the implementation - I'm second guessing this statement :)
> 
> One corner case is if a user would start kubelet and only then create a swap file. kubelet currently does not detect if that's happening. The admin can restart kubelet, but then I guess the already mounted volumes would lack the `noswap` option.
> 
> I'm not sure if there's any reason not to use the `noswap` option if it's supported. I even think that in the future, when it is widely supported enough, we can drop this logic and use the `tmpfs` option indefinitely.
> 
> WDYT?

Indeed, what if you attach a volume, create a swap partition and mount it through a DaemonSet?

Reply:
I don't know enough about the mount option to know if there's any downside / risk / side-effect to using it.

If we're 100% certain it's a no-op when swap is not enabled, that seems ok.

If not, I'd lean towards not changing behavior of volumes on hosts without swap, at least not in the first release.

I'll defer to node / storage folks who know the implications of the option better.

Reply:
FYI @mrunalp @SergeyKanzhelev

Reply:
Of course, I forgot about old kernels... so this _would_ fail for a kernel that is too old to support "noswap" as a mount option, per:

```console
root@vagrant:~# strace -s 65535 -e mount -- mount -t tmpfs -o noswap tmpfs /mnt
mount("tmpfs", "/mnt", "tmpfs", 0, "noswap") = -1 EINVAL (Invalid argument)
mount: /var/mnt: wrong fs type, bad option, bad superblock on tmpfs, missing codepage or helper program, or other error.
+++ exited with 32 +++
```

So we still would need to have the active tmpfs mount check, just in case...

Reply:
Until there are objections from anyone, the implementation remains to try to always use tmpfs if supported.

> So we still would need to have the active tmpfs mount check, just in case...

Indeed. The check is already in place, and we should drop it in the future IMO